src/text-editor.coffee

```diff
@@ -2587,6 +2587,16 @@ class TextEditor extends Model
       maintainClipboard = true
     return
 
+
+  # Essential: For each selection, only copy hightlighted text. Line is coptied if no selected text and cursor is at the beginning of the line
```

Comment:
I'd actually like to *not* add this to our public API right now. Could you change `Essential` -> `Private` here?

Also, there's a small typo: `hightlighted` -> `highlighted`.

Also, I don't think the last sentence of the comment is true; it doesn't look like there's any special behavior of the cursor is at the beginning of the line. Did you intend to implement that behavior? If not, we should just remove that sentence from the comment.